compiler/damlc/daml-compiler/src/DA/Daml/Compiler/DataDependencies.hs

```diff
@@ -267,11 +282,22 @@ generateSrcFromLf env = noLoc mod
     exports :: [LIE GhcPs]
     genState :: GenState
     ((exports, decls), genState) = runGen
-        ((,) <$> genExports <*> genDecls)
+        ((,) <$> genExports <*> genDecls <* genOrphanDepImports)
 
     modRefs :: Set ModRef
     modRefs = gsModRefs genState
 
+    genOrphanDepImports :: Gen ()
+    genOrphanDepImports = sequence_ $ do
+        Just LF.DefValue{dvalBinder=(_, ty)} <- [NM.lookup LFC.moduleImportsName . LF.moduleValues $ envMod env]
+        Just quals <- [LFC.decodeModuleImports ty]
+        LF.Qualified { LF.qualModule, LF.qualPackage } <- Set.toList quals
+        pure $ emitModRef ModRef
```

Comment:
What happens if we have both a reference to a type in a module as well as an orphan instance? It looks like we want a semigroup instance on `ModRefImpSpec` where `NoImpSpec` takes precedence and then change `emitModRef` to make sure that it overwrites.

Reply:
Hm, I think the only way to test that would involve solving #10772 first, since otherwise there's no way (that I can think of) for `"new-package" Main.daml` to import the instance and a type defined in module `"old-package" X.daml` without importing `X.daml` directly.

Reply:
in any case, I think this would only trigger a redundant import warning, but I'd have to check